### DISCUSSION TITLE
0.9.0 release

### DISCUSSION THREAD

**Discussion by baentsch** [open-quantum-safe/oqs-demos#243] [General]
Just tagged 0.9.0. The one addition to the [release documentation](https://github.com/open-quantum-safe/oqs-demos/releases/tag/0.9.0) that may be worth while is Chromium binaries. @pi-314159 would you want to contribute those (ideally for Windows and Linux)? If so please let me have download links. Please also let me know if I should put the binaries onto github or post your links.

**Comment by pi-314159** [comment]
Please download it here: https://www.raytonne.com/chromium117.tgz (MD5: 0f950591e3a1163fa53d47b095008896)

The tgz is 1.1G, so you may need to delete unnecessary files.

It's based on liboqs 0.8.0, but it works with 0.9.0 and current interop server settings. You may test it.

For Windows: We no longer have the installer... We only have a modified installer for internal use. 😄

**Reply by baentsch** [reply]
Thanks for the tgz. It is indeed awfully large. Would you be able to create something about 10% this size that still works? I am not sure github accepts binaries of this dimension... Let me dig in parallel whether I find a script for that from previous builds I did -- but of course I am not sure that would still work with v117.... Edit: OId Chromium cert import instructions are still valid: All fine. ~Last question: How can I import the test server CA.crt? Didn't find anything in the build instructions -- but then again, this is usage info and we don't seem to have a USAGE.md for Chromium...~ It may be worth while adding a HowTo link to the test website, though.

**Reply by baentsch** [reply]
Thanks very much: The download finally finished and I could successfully test the image. Very nice -- many more KEMs and the sig chains correctly displayed! The test server documentation referenced for Chromium is pretty outdated (wrong, actually): 

> When using the [OQS-enabled Chromium build](https://github.com/open-quantum-safe/oqs-demos/tree/main/chromium) to access this web site, be aware of the limitations concerning supported algorithms as documented [here](https://github.com/open-quantum-safe/boringssl/wiki/Implementation-Notes).

Could you please try to edit/correct the referenced Wiki article, @pi-314159 ? This would also ensure that those permissions are properly assigned to you.

**Reply by pi-314159** [reply]
@baentsch I made a small edit (changing line numbers). Overall, I don't see any problem with the BoringSSL Wiki. The `kDefaultGroups` in BoringSSL remains unchanged. For example, if we run `tool/bssl client -connect test.openquantumsafe.org:6011 -debug </dev/null`, we'll get an error because `hqc192` is not in the `kDefaultGroups`.

Smaller tgz: https://www.raytonne.com/chromium117-2.tgz (size: 275M; md5: `77ddc0c30c4fc3fe4099107b966d8f94`). I believe the file size limit in GitHub release is 2GB and hope 275M is acceptable.

I think we need to add a note about building Chromium: It's a painful process and we recommend people to build a "normal" chromium before trying to build the oqs-chromium... About USAGE.md: I think it's a good idea, but I don't have time to create one now. I think we can list common quantum safe crypto errors and how to fix them, link to the test website, and how to verify quantum safe algorithms are working properly etc.

**Reply by baentsch** [reply]
> Overall, I don't see any problem with the BoringSSL Wiki. The kDefaultGroups in BoringSSL remains unchanged. 

This surprises me: Your Chromium build reports to be able to connect perfectly to port 6011: 
![grafik](https://github.com/open-quantum-safe/oqs-demos/assets/57787676/7ef96d10-c05f-4fa8-be11-eeeffb459421)

The full list of client KEMs reported is 

> 0x9a9a:kyber512:kyber768:kyber1024:hqc128:hqc192:hqc256:bikel1:bikel3:p256_kyber512:p384_kyber768:p521_kyber1024:p256_hqc128:p384_hqc192:p521_hqc256:p256_bikel1:p384_bikel3:x25519_kyber768:frodo640aes:frodo640shake:frodo976aes:frodo976shake:frodo1344aes:frodo1344shake:p256_frodo640aes:p256_frodo640shake:p384_frodo976aes:p384_frodo976shake:p521_frodo1344aes:p521_frodo1344shake:x25519:secp256r1:secp384r1

> Smaller tgz:

Much appreciated: Works and is attached to release now.

USAGE.md: I'd be glad to give that a go -- once I understand what's the real client group list enabled (for Chromium) as per the above.

**Reply by pi-314159** [reply]
@baentsch Chromium supports it by default because it's enabled in oqs-changes.patch (here: https://github.com/open-quantum-safe/oqs-demos/blob/main/chromium/oqs-changes.patch#L445) . However, in BoringSSL it's not enabled by default. :) The list of algorithms supported by default is different in Chromium and BoringSSL.

**Reply by baentsch** [reply]
> The list of algorithms supported by default is different in Chromium and BoringSSL.

OK; Thanks for the pointer! One more reason to document a Chromium USAGE.md. PR coming soon...
